accounts/views.py

```diff
@@ -156,6 +162,109 @@ def get_queryset(self):
         return qs
 
 
+class UserView(generics.RetrieveUpdateAPIView):
+    """
+    get:
+    Return information about the logged in user.
+
+    update:
+    Update information about the logged in user.
+    You must specify all of the fields or use a patch request.
+
+    patch:
+    Update information about the logged in user.
+    Only updates fields that are passed to the server.
+    """
+
+    serializer_class = UserSerializer
+    permission_classes = [IsAuthenticated]
+    # lookup_field = "pennid"
+
+    def get_object(self):
+        return self.request.user
+
+
+class PhoneNumberViewSet(viewsets.ModelViewSet):
+    """
+    retrieve:
+    Return a single phone number with all information fields present.
+
+    list:
+    Return a list of phone numbers associated with current user.
+
+    create:
+    Add new unverified phone number.
+
+    update:
+    Update all fields.
+    You must specify all of the fields or use a patch request.
+
+    partial_update:
+    Update certain fields.
+    Only specify the fields that you want to change.
+
+    destroy:
+    Delete a phone number.
+    """
+
+    serializer_class = PhoneNumberSerializer
+    permission_classes = [IsAuthenticated]
+
+    def get_queryset(self):
+        return self.request.user.phone_numbers.all()
+
+    def destroy(self, request, *args, **kwargs):
+        is_primary = self.get_object().primary
+        self.get_object().delete()
+        if is_primary and len(self.get_queryset().filter(verified=True)) > 0:
+            next_number = self.get_queryset().filter(verified=True)[0]
+            next_number.primary = True
+            next_number.save()
+        return HttpResponse("Number successfully deleted", status=200)
+
+
+class EmailViewSet(viewsets.ModelViewSet):
+    """
+    retrieve:
+    Return a single email with all information fields present.
+
+    list:
+    Return a list of emails associated with current user.
+
+    create:
+    Add new unverified email.
+
+    update:
+    Update all fields.
+    You must specify all of the fields or use a patch request.
+
+    partial_update:
+    Update certain fields.
+    Only specify the fields that you want to change.
+
+    destroy:
+    Delete an email.
+    """
+
+    serializer_class = EmailSerializer
+    permission_classes = [IsAuthenticated]
+
+    def get_queryset(self):
+        return self.request.user.emails.all()
+
+    def destroy(self, request, *args, **kwargs):
+        if len(self.get_queryset().filter(verified=True)) < 2:
```

Comment:
In general, you shouldn't call len() on a queryset. You should use qs.count() instead because that makes a more efficient SQL query.